src/main/java/org/folio/orders/utils/HelperUtils.java

```diff
@@ -375,57 +406,52 @@ private static void validateCostPrices(Cost cost, CompositePoLine.OrderFormat or
       errors.add(ErrorCodes.COST_ADDITIONAL_COST_INVALID);
     }
 
-    double discount = defaultIfNull(cost.getDiscount(), 0d);
-    if (discount < 0d || (cost.getDiscountType() == Cost.DiscountType.PERCENTAGE && discount > 100d)
-      // validate that discount does not exceed total price
-      || (discount > 0d && cost.getDiscountType() == Cost.DiscountType.AMOUNT && calculateEstimatedPrice(cost) < 0d)) {
+    if (isDiscountExceedTotalPrice(cost)) {
       errors.add(ErrorCodes.COST_DISCOUNT_INVALID);
     }
+    return errors;
+  }
+
+  private static boolean isDiscountExceedTotalPrice(Cost cost) {
```

Comment:
(nitpick) The method checks also if discount is not negative which does not correspond to method name

Reply:
Renamed